Port to python3 [wip]

Benjaki2 opened the pull request:
## Description

Fixes #1682 
Update build to use `python3`

## Checklist
- [ ] update documentation

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [ ] I have read the [CONTRIBUTING](https://github.com/nasa-gibs/worldview/blob/master/.github/CONTRIBUTING.md) doc
- [ ] I have added necessary documentation (if applicable)
- [ ] I have added tests that prove my fix is effective or that my feature works (if applicable)
- [ ] Any dependent changes have been merged and published in downstream modules (if applicable)

## Further comments

used [2to3](https://docs.python.org/2/library/2to3.html) to initial convert but some other minor changes related to how text strings are now `unicode` by default

@nasa-gibs/worldview

ZachTRice commented on tasks/python3/processVectorStyles.py:
```diff
@@ -0,0 +1,63 @@
+    if input_file.endswith('.json'):
+        response_data = {}
+        vector_layer_filename = file;
+        vector_layer_id = vector_layer_filename.split(".", 1)[0]
```
I have an open PR that changes this line; It probably makes sense to get your work in first and I will rebase my PR on top of it and add the change to the python3 directory. https://github.com/nasa-gibs/worldview/pull/2013

ZachTRice commented on tasks/python3/processVectorData.py:
```diff
@@ -0,0 +1,63 @@
+    if input_file.endswith('.json'):
+        response_data = {}
+        vector_layer_filename = file;
+        vector_layer_id = vector_layer_filename.split(".", 1)[0]
```
I have an open PR that changes this line; It probably makes sense to get your work in first and I will rebase my PR on top of it and add the change to the python3 directory. https://github.com/nasa-gibs/worldview/pull/2013

Benjaki2 commented:
@ZachTRice @edplato I am second guessing keeping `python2` as an option. If we want to do this we will need to update the python code in two places whenever there is a change or we will need to rely on `2to3` to compile on the fly when code changes and with our current 2.7 code-base it doesn't compile perfectly out-of-the box I had to debug quite a bit. Does dropping 2.7 support so alright you y'all?

ZachTRice commented:
@Benjaki2 Yes, I think it makes sense to drop now that I have seen the implementation. Maintaining 2 versions is also creating some weird edges cases where a user might have python3 added to their path as `python`; I think the best approach is going to be to just upgrade and leave python 2.x behind.

mike-mcgann commented:
Drop 2.7

edplato commented:
I was just looking into this and saw that `Windows Build Tools` is still using `v2.7.x` because `v3.x.x` is not supported in `node-gyp` yet.

ZachTRice commented:
Looks like it is working in Travis with python 3

ZachTRice commented:
One thing to note is that the Travis setup uses the `choco install` of python3; if there happens to be a conflict on our local machines, we might want to check if this install differs in anyway from the official msi/exe.

Benjaki2 commented:
@ZachTRice I removed 2.x python scripts and removed the checks for legacy versions in the `pythonInstall`.  I also rebased this branch on develop and manually applied your changes to the `vectordata/vectorStyle` python scripts.

@edplato were you able to get this up and running on windows?
